pipelines/pipeline_katello_36_to_37_upgrade.yml

```diff
@@ -0,0 +1,156 @@
+  become: yes
+  vars:
+    foreman_installer_scenario: katello
+    foreman_installer_options_internal_use_only:
+      - "--foreman-admin-password {{ foreman_installer_admin_password }}"
+      - "--disable-system-checks"
+    foreman_installer_additional_packages:
+      - katello
+  roles:
+    - foreman_installer
+
+- hosts: pipeline-proxy-3.6-3.7-centos7
+  become: yes
+  vars:
+    foreman_proxy_content_server: pipeline-katello-3.6-3.7-centos7
+    foreman_installer_scenario: foreman-proxy-content
+    foreman_installer_options_internal_use_only:
+      - "--disable-system-checks"
+      - "--foreman-proxy-trusted-hosts {{ server_fqdn.stdout }}"
+      - "--foreman-proxy-trusted-hosts {{ ansible_nodename }}"
+      - "--foreman-proxy-foreman-base-url https://{{ server_fqdn.stdout }}"
+      - "--foreman-proxy-register-in-foreman true"
+      - "--foreman-proxy-oauth-consumer-key {{ oauth_consumer_key.stdout }}"
+      - "--foreman-proxy-oauth-consumer-secret {{ oauth_consumer_secret.stdout }}"
+      - "--foreman-proxy-content-certs-tar {{ foreman_proxy_content_certs_tar }}"
+      - "--foreman-proxy-content-parent-fqdn {{ server_fqdn.stdout }}"
+      - "--puppet-server-foreman-url https://{{ server_fqdn.stdout }}"
+    foreman_installer_additional_packages:
+      - foreman-installer-katello
+  roles:
+    - foreman_proxy_content
+    - foreman_installer
+
+- hosts: pipeline-katello-3.6-3.7-centos7
+  become: true
+  vars:
+    bats_tests:
+      - fb-test-katello.bats
+      - fb-content-katello.bats
+  roles:
+    - bats
+
+- hosts: pipeline-katello-3.6-3.7-centos7
+  become: true
+  vars:
+    katello_repositories_version: 3.7
+    foreman_repositories_version: 1.18
+    foreman_installer_upgrade: True
+    foreman_repositories_environment: staging
+    katello_repositories_environment: staging
+    foreman_installer_scenario: katello
+    foreman_installer_options_internal_use_only:
+      - "--disable-system-checks"
+  roles:
+    - foreman_repositories
+    - katello_repositories
+    - foreman_installer
+
+- hosts: pipeline-katello-3.6-3.7-centos7
+  become: true
+  vars:
+    bats_tests:
+      - fb-test-katello.bats
+      - fb-destroy-organization.bats
+      - fb-content-katello.bats
+      - fb-proxy.bats
+      - fb-finish.bats
+  roles:
+    - bats
+
+# Upgrade to Puppet 5
+# http://projects.theforeman.org/projects/foreman/wiki/Upgrading_from_Puppet_4_to_5#Prepare-And-Run-The-Installer
+- hosts:
+  - pipeline-katello-3.6-3.7-centos7
+  - pipeline-proxy-3.6-3.7-centos7
+  become: true
+  vars:
+    puppet_repositories_version: 5
+  roles:
+    - puppet_repositories
+
+- hosts:
+  - pipeline-katello-3.6-3.7-centos7
+  - pipeline-proxy-3.6-3.7-centos7
+  become: true
+  tasks:
+    - name: 'Update packages'
+      yum:
+        name: '*'
+        state: latest
+
+- hosts: pipeline-katello-3.6-3.7-centos7
+  become: true
+  vars:
+    foreman_installer_scenario: katello
+    foreman_installer_options_internal_use_only:
```

Comment:
Wait, the installer just handles the puppet upgrade automagically? You don't need `foreman_installer_upgrade: True`?

Reply:
No idea. I see this when the installer runs:

```
[ WARN 2018-06-11T19:12:20 verbose] -:puppet_version: 4.10.12                                                                                     
[ WARN 2018-06-11T19:12:20 verbose] +:puppet_version: 5.5.2 
```

What should I be looking for?

Reply:
Trying with that upgrade flag right now

Reply:
I'm going to wait until the upgrade related issues are fixed before looking into this more.